### ISSUE TITLE
incorrect return values in vocation:getRequiredSkillTries(skill, level)

### ISSUE THREAD

**Issue by Zbizu** [otland/forgottenserver#3217]
tl;dr: It's a rounding error that seems to be happening in vocation.cpp only

### Before creating an issue, please ensure:
- [x] This is a bug in the software that resides in this repository, and not a
      support matter (use https://otland.net/forums/support.16/ for support)
- [x] This issue is reproducible without changes to the C++ code in this repository

### Steps to reproduce (include any configuration/script required to reproduce)
1. print(Vocation(3):getRequiredSkillTries(SKILL_AXE, 100))
2. put actual formula instead of c++ one

**lua:** `print(50 * 1.2^(100 - 11))`
**wolframalpha:** type `50 * 1.2^(100 - 11)` in formula box (wolframalpha.com)
**mysql:** type `select FLOOR(50 * POW(1.2,(100 - 11))) as '' ` in phpmyadmin

### Expected behaviour
return value should be 557315218 or 557315218.7056

### Actual behaviour
return value in tfs is 557317189

### Environment
Windows 7, compilation mode 64-bit release

**Comment by Shawak** [comment]
The `float skillMultipliers[SKILL_LAST + 1] = {1.5f, 2.0f, 2.0f, 2.0f, 2.0f, 1.5f, 1.1f};` differ, you used 1.2 but tfs uses 2.0 fox axe.

**Comment by MillhioreBT** [comment]
I have been doing tests by compiling the server several times but the returned value is always wrong.

I did some tests outside the compiler and the result was correct, as shown in the following image:
![image](https://user-images.githubusercontent.com/28090948/99085551-b6640200-259e-11eb-8bc1-8bee734b9d29.png)

I think the problem lies in some configuration present in the compiler regarding the function: `std::pow`

**Comment by Zbizu** [comment]
I think it's more of a rounding/cast error

it returns correct value on doubles only:
```
#include <iostream>
#include<math.h>
using namespace std;

int main() {
	double a = 50.0;
	double b = 1.2;
	double c = 89.0;

	double result = a * pow(b, c);
	cout << result;
	cin.get();

	return 0;
}
```
environment: windows 7, codeblocks 16.01, c++11, gnu gcc

**Comment by Zbizu** [comment]
I tested it on double only and it returned correct values. However, when I reproduced tfs code, it showed incorrect value again so the mistake must be in the code itself.

```
#include <iostream>
#include<math.h>
#include<map>

using namespace std;

std::map<uint32_t, uint32_t> cacheSkill[2];

uint64_t getReqSkillTries(uint16_t level)
{
    float skillmult = 1.2f;
    uint32_t skillBase = 50;
	auto it = cacheSkill[1].find(level);
	if (it != cacheSkill[1].end()) {
		return it->second;
	}

	uint64_t tries = static_cast<uint64_t>(skillBase * pow(static_cast<double>(skillmult), level - 11));
	cacheSkill[1][level] = tries;
	return tries;
}

int main() {
    uint64_t result = getReqSkillTries(100);
	cout << result;
	cin.get();

	return 0;
}
```

**Comment by MillhioreBT** [comment]
It must be some compiler configuration, some decimal limiter or something like that, I have not yet started looking for a solution, but it is what I suppose.

**Comment by Zbizu** [comment]
when I use `double skillmult = 1.2;` instead of `float skillmult = 1.2f;` it works correctly

```
    double skillmult = 1.2;
    float skillmult2 = 1.2f;

    cout << skillmult << " " << skillmult2 << " " << (static_cast<double>(skillmult2) == skillmult) << endl;
```

output: 1.2 1.2 0

**Comment by EPuncker** [comment]
if someone want to test and/or open a PR
https://github.com/SaiyansKing/optimized_forgottenserver/commit/a5f525ac1799f0ff2ce4afa1aedba34fc2faaaa0
